src/Redux/Rocket/rocketSlice.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable */
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const url = 'https://api.spacexdata.com/v3/rockets';
+
+const getRocket = async () => {
+  const res = await fetch(url);
+  const rockets = await res.json();
+  return rockets;
+};
+
+export const fetchRockets = createAsyncThunk('rockets/fetchRockets', getRocket);
+const rocketSlice = createSlice({
+  name: 'rockets',
+  initialState: {
+    rockets: [],
+    status: 'Rocket_display',
+    error: null,
+  },
+  reducers: {
+    rocketsReserve: (state, action) => {
+      const newState = state.rockets.map((rocket) => {
+        if (rocket.id !== action.payload) return rocket;
+        return { ...rocket, reserved: !rocket.reserved };
+      });
+      state.rockets = newState;
+    },
+  },
+
+  extraReducers(builder) {
+    builder.addCase(fetchRockets.pending, (state) => {
+      state.status = 'loading';
+    });
+    builder.addCase(fetchRockets.fulfilled, (state, action) => {
+      state.status = 'succeeded';
+      const realRocket = [];
+      action.payload.map((rocket) => realRocket.push({
+        id: rocket.id,
+        name: rocket.rocket_name,
+        description: rocket.description,
+        image: rocket.flickr_images[1],
+        reserved: false,
```

Comment:
- [ ] As required for the project, kindly make sure tor the image of a rocket, you use the first image in the array of `flickr_images`. i.e. `image:` property of your rocket object should be update in order to get the 1st image of the array(`1st_image` = ` rocket.flickr_images[0]`)

1.